### PR TITLE
[Y-Cable][Broadcom] upgrade to support Broadcom Y-Cable API to release 1.2

### PR DESCRIPTION
**Pull request by vdahiya12**
This release goes in sync with the following firmware version of Broadcom Y cable, which is consistent with release 6
{
    "version_nic_active": "D203.9.D103.1",
    "version_nic_inactive": "D203.9.D103.1",
    "version_nic_next": "D203.9.D103.1",
    "version_peer_active": "D302.8",
    "version_peer_inactive": "D302.8",
    "version_peer_next": "D302.8",
    "version_self_active": "D302.8",
    "version_self_inactive": "D302.8",
    "version_self_next": "D302.8"
}

Signed-off-by: vaibhav-dahiya <vdahiya@microsoft.com>

#### Description
Basically a vendor specific implementation of abstract YCableBase class .
detailed design discussion can be found https://github.com/Azure/SONiC/pull/757/files

#### Motivation and Context
to support the Y-Cable API required to support Broadcom's Y-Cable.

#### How Has This Been Tested?
put the changes in PMON, and all API's seem to run OK.

### CODE REVIEW

**Review by lguohan — APPROVED** [review]

**Comment by lguohan** [comment]
can you fix the lgtm alerts.

**Comment by vdahiya12** [comment]
@lguohan this is ready for merge.

**Review by lguohan — APPROVED** [review]
